### PR TITLE
Test Fedora Messaging messages in CI

### PR DESCRIPTION
**Pull request by abompard**
This PR adds the infrastructure to send and receive Fedora Messaging messages in the CI environment, and test the messages that went through.

### CODE REVIEW

**Comment by abompard** [comment]
What can be added to this:
- add a wrapper to send messages from the CI tests
- add tests for the consumers

**Comment by abompard** [comment]
It looks like Jenkins is not running the integration tests, I don't know why. I'm looking into it but if you have an idea I'm all ears.

**Comment by abompard** [comment]
Ah, that's because of #3157

**Comment by abompard** [comment]
OK, the failures we have now are related to the DNF API breakage.

File: devel/ci/integration/rabbitmq/rabbitmqadmin
Comment (bowlofeggs) [inline devel/ci/integration/rabbitmq/rabbitmqadmin]: Can we link to where this file comes from?

Comment (bowlofeggs) [reply devel/ci/integration/rabbitmq/rabbitmqadmin]: Or, would it make more sense to package it as an RPM, or maybe to just have the CI script download it from an authoritative source?

File: devel/ci/integration/tests/fixtures/rabbitmq.py
Comment (bowlofeggs) [inline devel/ci/integration/tests/fixtures/rabbitmq.py]:
```suggestion
# Copyright © 2019 Red Hat, Inc.
```

File: devel/ci/integration/tests/fixtures/rabbitmq.py
Comment (bowlofeggs) [inline devel/ci/integration/tests/fixtures/rabbitmq.py]: For new code, I'd like us to start using type annotations instead of docblocks to mark the types of things. So the above would be like this:

```
def rabbitmq_container(docker_backend: conu.DockerBackend, docker_network: str) -> conu.DockerContainer:
    """
    Fixture preparing and yielding a RabbitMQ container.

    Args:
        docker_backend: The Docker backend (fixture).
        docker_network: The Docker network ID (fixture).
    Yields:
        The RabbitMQ container.
```

File: devel/ci/integration/tests/fixtures/rabbitmq.py
Comment (bowlofeggs) [inline devel/ci/integration/tests/fixtures/rabbitmq.py]: Let's add a docblock and type annotations here too.

File: devel/ci/integration/tests/test_messages.py
Comment (bowlofeggs) [inline devel/ci/integration/tests/test_messages.py:19]: Let's add a license/copyright header on this.

File: devel/ci/integration/tests/test_messages.py
Comment (bowlofeggs) [inline devel/ci/integration/tests/test_messages.py:34]: I wonder if we could use [asyncio.wait()](https://docs.python.org/3.6/library/asyncio-task.html#asyncio.wait) here so that we will wait *up to* a timeout but not always the length of the sleep here? This way we can react as the messages arrive, but also won't hang forever if they don't arrive. I once found a sleep statement in Bodhi that was causing its unit tests to take a good bit longer and got a pretty big boost by dropping it in d26d42a4468f07839ee649f4808c1ed45af7e382 ☺ What do you think?

**Review by bowlofeggs — CHANGES_REQUESTED** [review]

File: devel/ci/integration/rabbitmq/rabbitmqadmin
Comment (abompard) [reply devel/ci/integration/rabbitmq/rabbitmqadmin]: Yeah that can be done, it's actually served from the RabbitMQ server itself.

**Comment by abompard** [comment]
OK I've made some changes in the last commit (to be squashed before merge). I removed `rabbitmqadmin` because it's not actually necessary for what I was using it for.

I don't know how to address the `time.sleep()` issue, though. I don't think tests are run asynchronously in pytest. I could implement a listener as a fixture but that would bring the fedora-messaging stack into the test host, and then it would require tests to be run through Twisted. It may be doable but I'm not sure it's worth it.

**Comment by abompard** [comment]
I rebased the PR, I think this is ready for re-review.
The CI error is caused by an upstream bug I think.

File: devel/ci/integration/bodhi/fedora-messaging.toml
Comment (cverna) [inline devel/ci/integration/bodhi/fedora-messaging.toml]:
```suggestion
callback = "bodhi.server.consumers:Consumer"
```
This was changed for bodhi 4.0

**Comment by cverna** [comment]
@abompard  One small change needed otherwise LGTM

**Comment by abompard** [comment]
OK, I implemented your change @cverna , and I rebased and squashed the commits. This should be good for re-review. Thanks.

**Review by cverna — APPROVED** [review]
LGTM

File: devel/ci/integration/tests/test_messages.py
Comment (abompard) [reply devel/ci/integration/tests/test_messages.py:34]: I'm not sure how to do that. What event will trigger the completion of the future in `wait()`? I don't think we can bubble up that event from the container to the host, where the tests are run. Can we?

Comment (bowlofeggs) [reply devel/ci/integration/tests/test_messages.py:34]: Yeah I didn't really consider the complexity of detecting changes from within the container.

**Review by bowlofeggs — APPROVED** [review]
Thanks!

**Pull request merged**
